### ISSUE TITLE
Vivaldi 5.5.2805.38 never loads with 1.2.4 and resources b59f91583acbbc4030866a3626d2d9f7f1fb8b54

### ISSUE THREAD

**Issue by Aldaviva** [Aldaviva/VivaldiCustomLauncher#6]
1. Update to Vivaldi 5.5.2805.38
2. Exit Vivaldi
3. Run VivaldiCustomLauncher

- Vivaldi shows the full-window logo loading screen forever
- Not sure where error messages are being sent, possibly a log file or Developer Tools with `--debug-packed-apps`
- Restoring `bundle.js` and `background-common-bundle.js` from `vivaldi.7z` lets it load again, so the issue is not the custom CSS or JS files, just the modifications made to one of the existing JS files

**Comment by Aldaviva** [comment]
I wonder if an additional automated smoke test could try to apply tweaks and then assert that Vivaldi can actually load correctly. I am not sure how we can determine that it loaded properly though. Maybe loading a page from a local web server, or asserting that certain error strings don't appear in log files?

**Comment by Aldaviva** [comment]
From `--debug-packed-apps`:
```text
Uncaught SyntaxError: Missing initializer in const declaration
bundle.js
```

Sadly, there is no context for the offset in `bundle.js`, nor does this file even appear in DevTools, so I still don't know where the problem is.

---

When launched with `--enable-logging --v=1`, `%LOCALAPPDATA%\Vivaldi\User Data\chrome_debug.log` contains
```text
[9956:11940:1013/111352.870:ERROR:CONSOLE(1)] "Uncaught SyntaxError: Missing initializer in const declaration", source: chrome-extension://mpognobbkildjkofajifpdfhcoklimli/bundle.js (1)
```
A hypothetical smoke test could use these same arguments and look for `:ERROR:CONSOLE(` in that file after some time.

**Comment by Aldaviva** [comment]
The following line seems to only appear in `chrome_debug.log` if the startup was successful, and is not present when there is an uncaught exception in `bundle.js`.
```text
:VERBOSE1:CONSOLE(1)] "JS init startup: 287.68701171875 ms", source: chrome-extension://mpognobbkildjkofajifpdfhcoklimli/bundle.js (1)
```
It actually appears twice in a successful log file, probably because of background pages or service workers.

The successful log file doesn't contain `:ERROR:CONSOLE(`, so a test could wait until `JS init startup` appears in the log and then assert that `:ERROR:CONSOLE(` does not appear.

**Comment by Aldaviva** [comment]
I now have an automated test that checks for this sort of failure: https://github.com/Aldaviva/VivaldiCustomLauncher/actions/runs/3257775328/jobs/5349297863#step:15:49

**Comment by Aldaviva** [comment]
I think this was introduced in 1.2.4 by 9d2fc09. The `increaseMaximumTabWidth` tweak was trying to emit the string
```cs
"=4000/* Customized by Ben */"
```
however, the C# expression
```cs
'=' + 4000 + "/* Customized by Ben */"
```
coerces the char to int32 and evaluates the leftmost `+` expression using int32 addition instead of string concatenation, resulting in
```cs
"4061/* Customized by Ben */"
```
because the ASCII codepoint for `=` is 61 (0x3D), and it gets numerically added to 4000. This was the cause of the "Missing initializer in const declaration" uncaught exception in Vivaldi's error logs.

**Comment by Aldaviva** [comment]
Released in 1.2.5.
